src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -815,16 +828,19 @@ public UInt256 GetLatestPendingNonce(Address address)
         public event EventHandler<TxEventArgs>? RemovedPending;
         public event EventHandler<TxEventArgs>? EvictedPending;
 
-        public void Dispose()
+        public async ValueTask DisposeAsync()
         {
             if (_isDisposed) return;
             _isDisposed = true;
             _timer?.Dispose();
+            _cts.Cancel();
             TxPoolHeadChanged -= _broadcaster.OnNewHead;
             _broadcaster.Dispose();
             _headInfo.HeadChanged -= OnHeadChange;
             _headBlocksChannel.Writer.Complete();
             _transactions.Removed -= OnRemovedTx;
+
+            await _headProcessing;
```

Comment:
why awaiting something irrelevant when shutting down?

Reply:
Is potentially accessing state db so can throw exceptions if that is disposed and txpool still running

Reply:
set a flag and catch those exceptions in the background thread if flag is set?